src/object/obj_rpc.c

```diff
@@ -347,29 +341,25 @@ crt_proc_struct_daos_recx_ep_list(crt_proc_t proc,
 	if (list->re_nr == 0)
 		return 0;
 
-	switch (proc_op) {
-	case CRT_PROC_DECODE:
+	if (DECODING(proc_op)) {
 		D_ALLOC_ARRAY(list->re_items, list->re_nr);
 		if (list->re_items == NULL)
 			return -DER_NOMEM;
 		list->re_total = list->re_nr;
-		/* fall through to fill re_items */
-	case CRT_PROC_ENCODE:
-		rc = crt_proc_memcpy(proc, list->re_items,
-				     list->re_nr * sizeof(*list->re_items));
-		if (unlikely(rc && DECODING(proc_op))) {
+	}
+	rc = crt_proc_memcpy(proc, list->re_items,
```

Comment:
Looking at this code crt_prop_memcpy() should take a proc_op rather than a proc but that's out of scope for this PR.

Reply:
I though about this but this is not always possible. This function can be called with proc only in case of automatic serialization code generation. So, it required a lot of changes. I'm not ready for this now.

Reply:
Maybe there's a case for two functions here, particularly given how often this is called.  I think some of the error checking in that function is unnecessarily heavy as well.

Reply:
Maybe. I will think about this more. Anyway this work for different patch.